kong/client.go

```diff
@@ -164,6 +165,34 @@ func NewClient(baseURL *string, client *http.Client) (*Client, error) {
 	return kong, nil
 }
 
+// WithWorkspace return the client with a defined workspace
+func (c *Client) WithWorkspace(workspace string) *Client {
+	c.SetWorkspace(workspace)
+	return c
+}
+
+// SetWorkspace defines a workspace for the client
+func (c *Client) SetWorkspace(workspace string) {
+	c.workspace = workspace
+}
+
+//ClearWorkspace clear the workspace
+func (c *Client) ClearWorkspace() {
+	c.workspace = ""
+}
+
+// baseURL build the base URL from the rootURL and the workspace
+func (c *Client) baseURL() string {
+	if c.hasWorkspace() {
+		return c.rootURL + "/" + c.workspace
+	}
+	return c.rootURL
+}
```

Comment:
```suggestion
// baseURL build the base URL from the rootURL and the workspace
func (c *Client) baseURL(ws string) string {
	if len(ws) > 0 {
			return c.rootURL + "/" + ws
	}
	return c.rootURL
}
```

To avoid the need to read the workspace multiple times from different functions, accept the workspace name as a parameter, and expect the caller to read the workspace only once.